src/OpenProject/WebViewIntegration/OpenProjectInstanceValidator.cs

```diff
@@ -64,26 +64,30 @@ public static async Task<(bool isValid, string instanceBaseUrl)> IsValidOpenProj
 
     private static string GetInstanceUrl(string instanceNameOrUrl)
     {
-      if (Uri.TryCreate(instanceNameOrUrl, UriKind.Absolute, out var _)
-        && Regex.IsMatch(instanceNameOrUrl.ToLower(), "^https?://"))
+      const string apiPathSuffix = "/api/v3";
+      bool hasApiSuffix = instanceNameOrUrl.TrimEnd('/').EndsWith(apiPathSuffix, StringComparison.InvariantCultureIgnoreCase);
+
+      string appendSuffix(string uri)
       {
-        return instanceNameOrUrl;
+        string suffix = hasApiSuffix ? "" : apiPathSuffix;
```

Comment:
I'd prefer `var` instead of `string` for consistency with the rest of the codebase, also for some other variables below.

Reply:
Will do. Do we use var everywhere? or only when it is evident? what are the guidelines you used?

Reply:
My guideline actually is to use `var` always. It's so rare, in my experience, that it's not obvious. And even then, it's usually for types no one knows by heart anyway. Long names don't provide a value in the IDE, and hardly any in text editors (or GitHub, which even now has code navigation), so, yeah, if you want to do it like me then you always use `var`. And imho it's more pretty, the other way always reminds me of Java😀